packages/pilot/src/containers/Settings/User/index.js

```diff
@@ -56,18 +48,44 @@ const UserSettings = ({
 }) => {
   const [tabIndex, setTabIndex] = useState(0)
   const history = useHistory()
+  const [tabIndexByName, setTabIndexByName] = useState({
+    accountInfo: 0,
+    bankingInfo: 1,
+    password: 3,
+    teamInfo: 2,
+  })
+  const changeTab = (newTabIndex) => {
+    setTabIndex(newTabIndex)
+  }
+
+  const hideBankAccount = isPaymentLink(company)
+  const tabs = [
+    'pages.settings.user.tabs.account_info',
+    'pages.settings.user.tabs.banking_info',
+    'pages.settings.user.tabs.team_info',
+    'pages.settings.user.tabs.change_password',
+  ]
 
   useEffect(() => {
     if (history.location.state) {
       const tabName = history.location?.state?.tab
       setTabIndex(tabIndexByName[tabName])
     }
-  }, [history.location.state])
+  }, [history.location.state, tabIndexByName])
 
-  const changeTab = (newTabIndex) => {
-    setTabIndex(newTabIndex)
-  }
+  useEffect(() => {
+    if (hideBankAccount) {
+      setTabIndexByName({
+        accountInfo: 0,
+        password: 2,
+        teamInfo: 1,
+      })
+    }
+  }, [company, hideBankAccount])
 
+  if (hideBankAccount) {
+    tabs.splice(1, 1)
```

Comment:
Não seria melhor inserir a tab de banking info, caso seja necessário? ao invés de tirar. Pq da forma que tá, caso os indíces mudem e a gente esqueça de atualizar esse splice, pode acabar removendo uma aba errada.

acho que algo assim:

```suggestion
    tabs.splice(tabIndexByName.bankingInfo, 0, 'pages.settings.user.tabs.banking_info')
```

e lá da definição de tabs, tirar a linha do banking info do array.